clients/rancher/client.go

```diff
@@ -68,16 +68,35 @@ func NewClient(bearerToken string, session *session.Session) (*Client, error) {
 	if bearerToken == "" {
 		bearerToken = rancherConfig.AdminToken
 	}
+	// rancherConfig.AdminToken = bearerToken
 
 	c := &Client{
 		RancherConfig: rancherConfig,
 		Flags:         &environmentFlags,
 	}
 
-	session.CleanupEnabled = *rancherConfig.Cleanup
+	return newClient(c, bearerToken, rancherConfig, session)
+}
+
+// NewClientForConfig is the constructor for initializing a rancher Client for the given config and session.
+func NewClientForConfig(bearerToken string, config *Config, session *session.Session) (*Client, error) {
```

Comment:
in what case would we want to send in the bearer token separately tho? why not retrieve it from the rancher config itself?

Reply:
The bearerToken is used for setting up the restConfig, and the managemnent, v1, wrangler, cli and catalog clients. It allows for things like the `Client.AsUser(user)` function to utilize the rancher client as a non-admin via a non-admin bearertoken that can be passed to the function.

Tests like `tests/v2/integration/projects/project_user_test.go` and actions like `tests/v2/actions/rbac/rbac.go` as well as various other places utilize the `Client.AsUser(user)` function.

Reply:
got it. In that case do we want to add a check to see if a bearer token is empty and if it is, we can retrieve it from the rancher config itself - similar to how we have it for `NewClient`. will be useful for admin users to not have to send in the token every time?

Reply:
Good call! Adding that in now